plugins/last-fm/back.js

```diff
@@ -0,0 +1,192 @@
+}
+
+const cleanupArtistName = (config, artist) => {
+	// removes the suffixes of the artist name for more recognition by last.fm
+	let { suffixesToRemove } = config;
+	if (suffixesToRemove === undefined) {
+		suffixesToRemove = defaultSettings.suffixesToRemove;
+		config.suffixesToRemove = suffixesToRemove;
+		setOptions('last-fm', config);
+	}
+	for (suffix of suffixesToRemove) {
+		artist = artist.replace(suffix, '');
+	}
+	return artist
+}
+
+const createFormData = params => {
+	// creates the body for in the post request
+	let formData = new URLSearchParams();
+	for (key in params) {
+		formData.append(key, params[key]);
+	}
+	return formData;
+}
+const createQueryString = (params, api_sig) => {
+	// creates a querystring
+	const queryData = [];
+	params.api_sig = api_sig;
+	for (key in params) {
+		queryData.push(`${key}=${params[key]}`);
+	}
+	return '?'+queryData.join('&');
+}
+
+const createApiSig = (params, secret) => { 
+	// this function creates the api signature, see: https://www.last.fm/api/authspec
+	let keys = [];
+	for (key in params) {
+		keys.push(key);
+	}
+	keys.sort();
+	let sig = '';
+	for (key of keys) {
+		if (String(key) === 'format')
+			continue
+		sig += `${key}${params[key]}`;
+	}
+	sig += secret;
+	sig = md5(sig);
+	return sig;
+}
+
+const createToken = async ({ api_key, api_root, secret }) => {
+	// creates and stores the auth token
+	data = {
+		method: 'auth.gettoken',
+		api_key: api_key,
+		format: 'json'
+	};
+	let api_sig = createApiSig(data, secret);
+	let response = await fetch(`${api_root}${createQueryString(data, api_sig)}`);
+	response = await response.json();
+	return response?.token;
+}
+
+const authenticate = async config => {
+	// asks the user for authentication
+	config.token = await createToken(config);
+	setOptions('last-fm', config);
+	open(`https://www.last.fm/api/auth/?api_key=${config.api_key}&token=${config.token}`);
+	return config;
+}
+
+const getAndSetSessionKey = async config => {
+	// get and store the session key
+	data = {
+		api_key: config.api_key,
+		format: 'json',
+		method: 'auth.getsession',
+		token: config.token,
+	};
+	api_sig = createApiSig(data, config.secret);
+	res = await fetch(`${config.api_root}${createQueryString(data, api_sig)}`);
+	res = await res.json();
+	if (res.error)
+		await authenticate(config);
+	config.session_key = res?.session?.key;
+	setOptions('last-fm', config);
+	return config;
+}
+
+
+const addScrobble = async (songInfo, config) => {
+	// this adds one scrobbled song to last.fm
+	if (!config.session_key)
+		await getAndSetSessionKey(config);
+	data = {
+		track: songInfo.title,
+		artist: songInfo.artist,
+		api_key: config.api_key,
+		sk: config.session_key,
+		format: 'json',
+		method: 'track.scrobble',
+		timestamp: ~~((Date.now() - songInfo.elapsedSeconds) / 1000),
+		duration: songInfo.songDuration,
+	};
+	data.api_sig = createApiSig(data, config.secret);
+	axios.post('https://ws.audioscrobbler.com/2.0/', createFormData(data))
```

Comment:
nit: node-fetch is already there and could be used (but axios is well-known and widely used so no big deal and no need to change it)

Reply:
Ah, didn't know you could post data with node-fetch, removed axios, because there is no need for an extra dependency :)